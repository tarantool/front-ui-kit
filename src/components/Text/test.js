import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from './index';

it('Text renders correctly', () => {
  const tree = renderer.create(
    <>
      <Text variant='h1' title='Text title'>Heading 1 - 24px, 38px, 600, spacing 0.48px</Text>
      <Text variant='h2'>Heading 2 - 20px, 28px, 600, spacing 0.4px</Text>
      <Text variant='h3'>Heading 3 - 16px, 24px, 600, spacing 0.32px</Text>
      <Text variant='h4'>Heading 4 - 16px, 24px, 600, spacing 0.32px</Text>
      <Text variant='h5'>Heading 5 - 12px, 22px, 400, spacing 0.24px</Text>
      <Text variant='p'>paragraph - 12px, 20px, 400, spacing: 0.28px</Text>
      <Text>basic - 14px, 22px, 400</Text>
      <Text variant='code'>code - 14px, 22px, 400</Text>
      <Text variant='h3' tag='span'>H3 styled span</Text>
      <Text variant='h4' tag='span'>H4 styled span</Text>
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
