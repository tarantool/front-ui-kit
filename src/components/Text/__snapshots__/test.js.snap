// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
Array [
  <h1
    className="css-1b2tio5"
    title="Text title"
  >
    Heading 1 - 24px, 38px, 600, spacing 0.48px
  </h1>,
  <h2
    className="css-1w3lc29"
  >
    Heading 2 - 20px, 28px, 600, spacing 0.4px
  </h2>,
  <h3
    className="css-1ayss2c"
  >
    Heading 3 - 16px, 24px, 600, spacing 0.32px
  </h3>,
  <h4
    className="css-115iv0u"
  >
    Heading 4 - 16px, 24px, 600, spacing 0.32px
  </h4>,
  <h5
    className="css-1lh30e5"
  >
    Heading 5 - 12px, 22px, 400, spacing 0.24px
  </h5>,
  <p
    className="css-1w6kemg"
  >
    paragraph - 12px, 20px, 400, spacing: 0.28px
  </p>,
  <span
    className="css-1ggp6x0"
  >
    basic - 14px, 22px, 400
  </span>,
  <code
    className="css-19aciib"
  >
    code - 14px, 22px, 400
  </code>,
  <span
    className="css-1ayss2c"
  >
    H3 styled span
  </span>,
  <span
    className="css-115iv0u"
  >
    H4 styled span
  </span>,
]
`;
